mailbox/backup/src/test/java/org/apache/james/mailbox/backup/ZipEntryTypeExtraFieldTest.java

```diff
@@ -0,0 +1,197 @@
+
+import com.google.common.base.Charsets;
+import nl.jqno.equalsverifier.EqualsVerifier;
+import nl.jqno.equalsverifier.Warning;
+
+public class ZipEntryTypeExtraFieldTest {
+    private static final byte[] ZERO_AS_BYTE_ARRAY = {0, 0, 0, 0, 0, 0, 0, 0};
+    private static final byte[] ONE_AS_BYTE_ARRAY = {1, 0, 0, 0, 0, 0, 0, 0};
+    private static final byte[] TWO_AS_BYTE_ARRAY = {2, 0, 0, 0, 0, 0, 0, 0};
+    private static final byte[] THREE_AS_BYTE_ARRAY = {3, 0, 0, 0, 0, 0, 0, 0};
+    private static final byte[] FOUR_AS_BYTE_ARRAY = {4, 0, 0, 0, 0, 0, 0, 0};
+
+    private EntryTypeExtraField testee;
+
+    @BeforeEach
+    void setUp() {
+        testee = new EntryTypeExtraField();
+    }
+
+    @Test
+    void shouldMatchBeanContract() {
+        EqualsVerifier.forClass(EntryTypeExtraField.class)
+            .suppress(Warning.NONFINAL_FIELDS)
+            .verify();
+    }
+
+    @Test
+    void getLocalFileDataLengthShouldReturnIntegerSize() {
+        assertThat(testee.getLocalFileDataLength().getValue())
+            .isEqualTo(Long.BYTES);
+    }
+
+    @Test
+    void getCentralDirectoryLengthShouldReturnIntegerSize() {
+        assertThat(testee.getCentralDirectoryLength().getValue())
+            .isEqualTo(Long.BYTES);
+    }
+
+    @Test
+    void getHeaderIdShouldReturnSpecificStringInLittleEndian() {
+        ByteBuffer byteBuffer = ByteBuffer.wrap(testee.getHeaderId().getBytes())
+            .order(ByteOrder.LITTLE_ENDIAN);
+        assertThat(Charsets.US_ASCII.decode(byteBuffer).toString())
+            .isEqualTo("aq");
+    }
+
+    @Test
+    void getLocalFileDataDataShouldThrowWhenNoValue() {
+        assertThatThrownBy(() -> testee.getLocalFileDataData())
+            .isInstanceOf(RuntimeException.class);
+    }
+
+    @Test
+    void getLocalFileDataDataShouldReturnZeroWhenZero() {
+        byte[] actual = new EntryTypeExtraField(ZipEntryType.MAILBOX).getLocalFileDataData();
+        assertThat(actual).isEqualTo(ONE_AS_BYTE_ARRAY);
+    }
+
+
+    @Test
+    void getLocalFileDataShouldReturnValueInLittleIndianWhenMailbox() {
+        byte[] actual = new EntryTypeExtraField(ZipEntryType.MAILBOX).getLocalFileDataData();
+        assertThat(actual).isEqualTo(ONE_AS_BYTE_ARRAY);
+    }
+
+    @Test
+    void getLocalFileDataShouldReturnValueInLittleIndianWhenMailboxAnnotationDir() {
+        byte[] actual = new EntryTypeExtraField(ZipEntryType.MAILBOX_ANNOTATION_DIR).getLocalFileDataData();
+        assertThat(actual).isEqualTo(TWO_AS_BYTE_ARRAY);
+    }
+
+    @Test
+    void getLocalFileDataShouldReturnValueInLittleIndianWhenMailboxAnnotation() {
+        byte[] actual = new EntryTypeExtraField(ZipEntryType.MAILBOX_ANNOTATION).getLocalFileDataData();
+        assertThat(actual).isEqualTo(THREE_AS_BYTE_ARRAY);
+    }
+
+    @Test
+    void getLocalFileDataShouldReturnValueInLittleIndianWhenMessage() {
+        byte[] actual = new EntryTypeExtraField(ZipEntryType.MESSAGE).getLocalFileDataData();
```

Comment:
s/DataData/Data/ ?

Reply:
Was there before but i will happily rename it.

Reply:
Can't be renamed it's in a jar : org.apache.commons.compress.archivers.zip.ZipExtraField